ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/SentenceHighlightingQATranslator.java

```diff
@@ -47,149 +65,332 @@
- * - 0 indicates a non-relevant sentence
  * 
- * This format can be customized by overriding the isRelevantPrediction method.
+ * This translator processes input for semantic sentence highlighting models that identify
+ * relevant sentences within a context document based on a user query or question.
+ * 
+ * The translator performs the following key functions:
+ * 1. Tokenizes the question and context using Hugging Face tokenizer
+ * 2. Segments the context into sentences 
+ * 3. Maps tokens to their corresponding sentence IDs
+ * 4. Handles chunking for long contexts that exceed the model's maximum token length
+ * 5. Processes model outputs to identify and highlight sentences that answer the question
+ * 
+ * The highlighted sentences are returned with their text and position information within
+ * the original context, which allows for easy visualization and extraction of relevant
+ * information from the document.
  */
 @Log4j2
 @Getter
 @Builder
 public class SentenceHighlightingQATranslator implements ServingTranslator {
     /**
-     * Default relevance value that indicates a sentence is relevant.
-     * By default, 1 means relevant and 0 means not relevant.
-     * The method specifically checks for equality with RELEVANT_VALUE (1) to determine relevance.
-     */
-    private static final long RELEVANT_VALUE = 1L;
-
-    /**
-     * Determines if a prediction value indicates a relevant sentence.
-     * 
-     * @param predictionValue The prediction value from the model
-     * @return true if the prediction indicates a relevant sentence, false otherwise
+     * This translator works with the semantic sentence highlighting model, which returns
+     * sentence indices directly rather than binary relevance scores.
      */
-    protected boolean isRelevantPrediction(long predictionValue) {
-        return predictionValue == RELEVANT_VALUE;
-    }
 
     @Builder.Default
     private final SentenceSegmenter segmenter = new DefaultSentenceSegmenter();
 
     private HuggingFaceTokenizer tokenizer;
 
+    private final MLModelConfig modelConfig;
+
+    /**
+     * Helper method to read a value from allConfig with a default fallback
+     * @param <T> The type of value to read (String, Integer, Boolean)
+     * @param key The config key to read
+     * @param defaultValue The default value to use if key not found or parsing fails
+     * @param valueType The class of the value type for type safety
+     * @return The value from config or default if not found
+     */
+    private <T> T readFromModelAllConfig(String key, T defaultValue, Class<T> valueType) {
+        if (modelConfig == null || modelConfig.getAllConfig() == null) {
+            return defaultValue;
+        }
+
+        try (XContentParser parser = JsonXContent.jsonXContent.createParser(null, null, modelConfig.getAllConfig())) {
+            Map<String, Object> configMap = parser.map();
+            Object value = configMap.get(key);
```

Comment:
when this key is not found, it will throw exception here and catch later, you can check if the key exist first

Reply:
I dont think there will have key not found case here, since `all_config` are passed in with json key value pair format, it already has validation earlier.

Reply:
isn't it this ConfigMap is already get from all_config?
modelConfig.getAllConfig() 

this is the parsing the inner key in `all_config`

Reply:
When key is not found from user deployed model, then it'll return default value instead of throw exception.
```
  Object value = configMap.get(key);

  if (value == null) {
      return defaultValue;
  }
```